features/FEATURE_BLE/ble/services/HealthThermometerService.h

```diff
@@ -72,10 +72,8 @@ class HealthThermometerService {
     *
     */
     void updateTemperature(float temperature) {
-        if (ble.getGapState().connected) {
```

Comment:
I guess this could be replaced by getConnectionCount()?

Reply:
Good point. Unfortunately getConnectionCount() is not available anymore (reverted soon after it was first introduced), but a few deprecation notices still points to that. I guess it needs to be fixed as well.

Reply:
This one removed getConnectionCount(): https://github.com/ARMmbed/mbed-os/commit/61536c3f6429eda66eb6ede5106d1601c50c10f5

Reply:
I don't know why it was removed. The variable backing it is still functional. @pan-?

Reply:
I'm not sure it was intentionally removed; just a mistake when non backward compatible changes were reverted. #8738 has been wild.

Reply:
So that's actually not required and application code doesn't have to ensure that it is connected before calling updateTemperature.